streams/src/test/java/org/apache/kafka/streams/state/internals/CompositeReadOnlySessionStoreTest.java

```diff
@@ -111,10 +112,40 @@ public void shouldThrowInvalidStateStoreExceptionOnRebalance() throws Exception
         store.fetch("a");
     }
 
-    @Test(expected = InvalidStateStoreException.class)
-    public void shouldThrowInvalidStateStoreExceptionIfFetchThrows() throws Exception {
+    @Test
+    public void shouldThrowInvalidStateStoreExceptionIfUnderlyingSessionStoreFetchThrows() {
+        underlyingSessionStore.setOpen(false);
+        try {
+            underlyingSessionStore.fetch("key");
+            fail("Should have thrown InvalidStateStoreException with underlying session store");
+        } catch (InvalidStateStoreException e) { }
+    }
+
+    @Test
+    public void shouldThrowInvalidStateStoreExceptionIfSessionFetchThrows() {
         underlyingSessionStore.setOpen(false);
-        underlyingSessionStore.fetch("key");
+        try {
+            sessionStore.fetch("key");
```

Comment:
We should also add `Objects.requireNonNull(..)` in `CachingSessionStore.fetch(..)` as the NPE is not thrown directly by the code under test